### ISSUE TITLE
What is the reasoning for a Request FSA with error flag? (vs a Failure FSA)

### ISSUE THREAD

**Issue by barrystaes** [agraboso/redux-api-middleware#44]
As i understand now - when a `REQUEST` FSA gets fired, this can result in a:
- `REQUEST` FSA with error flag (for example when the API request cant be made)
- `SUCCESS` FSA (for example for API response 200 OK)
- `FAILURE` FSA (for example for API response 404 Not Found)

But in my view one `REQUEST` FSA should result in (only and) exactly one FSA, either `SUCCESS` or `FAILURE`.

I haven't actually used it, but i dont really understand the first one. Whats the argument for not returning a `FAILURE` FSA in that case?

**Comment by agraboso** [comment]
@barrystaes This is an issue that I struggled with &mdash; see [this comment](https://github.com/agraboso/redux-api-middleware/issues/15#issuecomment-148865031) of mine. The solution that I came up with might not be to everyone's taste nor even the most logical one. I think a thorough discussion is in order before we push out a v2.

**Comment by agraboso** [comment]
Pinging #26, since it is relevant to this issue.

**Comment by aris-b** [comment]
I am also confused. When I get an API request error like this : 
```
{ 
 [RequestError: request to https://xxxxx failed, reason: self signed certificate]
 name: 'RequestError',
 message: 'request to https://xxxx failed, reason: self signed certificate' },
 meta: undefined,
 error: true }
```
The `FAILURE` FSA is not triggered. I see this error by logging on `REQUEST` FSA. 
So my question is, when actually `FAILURE` FSA  gets triggered ?

**Comment by danludwig** [comment]
Found this when searching for how to customize the action payload when a REQUEST FSA is dispatched with an error flag. Seems you can't? For example, if I attach a `key` property to the initial REQUEST payload, there is no way to correlate that key value back to the request that failed?

**Comment by barrystaes** [comment]
> So my question is, when actually FAILURE FSA gets triggered ?
@aris-b Short answer: when the response is not 200 OK. If the request fails, expect a `REQUEST` FSA with `error:true`. See [point 3 of lifecycle in readme.md](https://github.com/agraboso/redux-api-middleware/blob/e72ac27801cab4f5c5cedfaa8d1787e0ffce6f5b/README.md#lifecycle)

@danludwig Have you looked into using the `meta` property, which is there for the purpose of adding metadata to use later.

**Comment by iamandrewluca** [comment]
Fixed in #175
